### ISSUE TITLE
Match `jax.grad`/ `jax.value_and_grad`

### ISSUE THREAD

**Issue by ASEM000** [vvvm23/mezo-jax#1]
Hello, I wonder if the transformed function signature can be unchanged to match jax counterpart and act as a drop-in replacement without having to change the loss function?

**Comment by vvvm23** [comment]
Hi, I thought about doing this but decided against it as a deliberate design choice. For one, the gradient that is returned is actually a single scalar value, whereas the jax counterparts will return a pytree of gradients. Hence, it can't be used as a drop-in replacement so I gave it a different name. Hope that helps!

**Comment by ASEM000** [comment]
Noted, thanks

btw I think there is a minor typo in  the readme (=->+)

```python

def fn(params, x):
    y = x @ params['w'] = params['b']
    return (y**2).mean()
```

**Comment by vvvm23** [comment]
Good catch, thanks!
